server/aesdsocket.c

```diff
@@ -0,0 +1,232 @@
+                close(sockfd);
+                closelog();
+                exit(EXIT_FAILURE);
+            }
+            else if (child_pid > 0)
+            {
+                exit(EXIT_SUCCESS);
+            }
+            else
+            {
+                // This is the child. Convert this to a daemon.
+                if (setsid() == -1)
+                {
+                    syslog(LOG_ERR, "Error creating new session");
+                    close(sockfd);
+                    closelog();
+                    exit(EXIT_FAILURE);
+                }
+                chdir("/");
+
+                // Taken from Linux System Programming - Robert Love
+                /* redirect fd's 0,1,2 to /dev/null */
+                open("/dev/null", O_RDWR);
+                /* stdin */
+                dup(0);
+                /* stdout */
+                dup(0);
+                /* stderror */
+            }
+        }
+    }
+    status = listen(sockfd, BACKLOG);
+    if (status != 0)
+    {
+        syslog(LOG_ERR, "listen() error: %s\n", strerror(errno));
+    }
+
+    socklen_t client_address_size = sizeof(client_address);
+    while (true) // run infinitely until and error occurs or terminated by sigint.
+    {
+        int cfd = accept(sockfd, (struct sockaddr *)&client_address, &client_address_size);
+
+        if (cfd == -1)
+        {
+            syslog(LOG_ERR, "accept() error : %s\n", strerror(errno));
+        }
+
+        else
+        {
+            char peer_ip[INET6_ADDRSTRLEN];
+            char rec_buff[BUFF_SIZE];
+            rec_buff[BUFF_SIZE - 1] = 0; // Make the buffer null terminated for strchr to work correctly.
+
+            inet_ntop(client_address.ss_family, get_in_addr((struct sockaddr *)&client_address), peer_ip, INET6_ADDRSTRLEN);
+            syslog(LOG_USER, "Accepted conncetion from : %s\n", peer_ip);
+            
+            int log_fd = open(log_path, O_CREAT | O_APPEND | O_RDWR, S_IRWXU | S_IRGRP | S_IROTH);
+            int num_read_bytes = 0;
+            int num_write_bytes = 0;
+            
+            if (log_fd == -1)
+            {
+                syslog(LOG_ERR, "open() error for log file : %s\n", strerror(errno));
+                close(cfd);
+                break;
+            }
+            else
+            {
+                bool packet_complete = false;
+                while (!packet_complete)
+                {
+                    num_read_bytes = recv(cfd, rec_buff, BUFF_SIZE - 1, 0);
+                    if (num_read_bytes == -1)
+                    {
+                        syslog(LOG_ERR, "recv() error : %s\n", strerror(errno));
+                        close(cfd);
+                        break;
+                    }
+                    else if (num_read_bytes == 0 || (strchr(rec_buff, '\n') != NULL))
+                    {
+                        packet_complete = true;
+                    }
+                    num_write_bytes = write(log_fd, rec_buff, num_read_bytes);
```

Comment:
@parthkharade this will write to the buffer before the packet is complete which may cause issues with future assignments.